app/controllers/games_controller.rb

```diff
@@ -78,4 +80,17 @@ def new_game?
     flash[:alert] = "Sorry, you have to join the game first"
     redirect_to games_path
   end
+
+  # Check if game is in checkmate
+  # If so, present message indicating winner
+  def checkmate?
+    game = Game.find(params[:id])
+    return unless game.determine_checkmate
+
+    if game.turn_number.even?
+      game.update_attributes(winning_player_id: game.black_player_id)
+    elsif game.turn_number.odd?
+      game.update_attributes(winning_player_id: game.white_player_id)
```

Comment:
I'm going to change line 91 and 93 to `update` just because `update_attributes` is deprecated. shouldn't change anything in the code though!